src/lib/utils/os_utils.cpp

```diff
@@ -614,6 +615,25 @@ void OS::page_named(void* page, size_t size) {
 #endif
 }
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void OS::set_thread_name(std::thread& thread, std::string& name) {
+   #if defined(BOTAN_TARGET_OS_IS_LINUX) || defined(BOTAN_TARGET_OS_IS_FREEBSD)
+   static_cast<void>(pthread_setname_np(thread.native_handle(), name.c_str()));
+   #elif defined(BOTAN_TARGET_OS_IS_OPENBSD)
+   static_cast<void>(pthread_set_name_np(thread.native_handle(), name.c_str()));
+   #elif defined(BOTAN_TARGET_OS_IS_NETBSD)
+   static_cast<void>(pthread_set_name_np(thread.native_handle(), "%s", const_cast<char*>(name.c_str())));
+   #elif defined(BOTAN_TARGET_OS_HAS_WIN32) && defined(BOTAN_BUILD_COMPILER_IS_MSVC)
+   // Using SetThreadDescription from Win10
+   #else
+   // TODO other possible oses ?
+   // macOs does not seem to allow to name threads other than the current one.
+   BOTAN_UNUSED(thread);
+   BOTAN_UNUSED(name);
```

Comment:
You can pass multiple args to a single invocation of `BOTAN_UNUSED` - `BOTAN_UNUSED(thread, name)`